src/CommandLine/ParserResult.cs

```diff
@@ -159,21 +168,12 @@ public bool Equals(Parsed<T> other)
     /// <typeparam name="T">The type with attributes that define the syntax of parsing rules.</typeparam>
     public sealed class NotParsed<T> : ParserResult<T>, IEquatable<NotParsed<T>>
     {
-        private readonly IEnumerable<Error> errors;
 
-        internal NotParsed(TypeInfo typeInfo, IEnumerable<Error> errors)
-            : base(ParserResultType.NotParsed, typeInfo)
+        internal NotParsed(IEnumerable<Error> errors, TypeInfo typeInfo)

```

Comment:
Is there a reason to switch the parameters of ctor? It's better to keep internal API the same.

Reply:
I was trying to be have the API be consistent across the Hierarchy.
`ParseResult(value/error, type)` matches `Parsed(value, type)` and `NotParsed(error, type)`

I thought it made it easier to see the relation between the constructors better, but I can revert it if you prefer having the parameter orders not match.

Reply:
I see, but keeping test without change is very necessary. It proves no break change.

Reply:
I'm not sure I understand your comment. Are you saying that there may be other places using the "internal" constructor outside of this library? That shouldn't be possible unless someone is doing very weird reflection:

https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/internal

the change should only affect the internals of this library, which includes the test cases.

Reply:
Internals are used by other PRs and test suit.
There are some PRs are still open and sure that will effect their work when merged and cause confliction when merging.

Reply:
I will revert those changes then.

Reply:
Thanks for the good work.